.github/workflows/kcp-image.yaml

```diff
@@ -10,6 +10,10 @@ on:
       - 'release-*'
     tags:
     - 'v*'
+  pull_request:
+    branches:
+      - main
+      - 'release-*'
```

Comment:
is this just for testing?

Reply:
I'd be inclined to keep it, so as to guarantee PRs do not break the image build for some reasons. In that case, the image is only built, not pushed, which I think is what we want for PRs:

https://github.com/kcp-dev/kcp/blob/c14bdeef94c4f769a2b4ae31eb293505c25c90b5/.github/workflows/kcp-image.yaml#L54

Reply:
With 21min it is much faster now. So I am less concerned this blocks our github action queue.